osmdroid-shape/src/main/java/org/osmdroid/shape/ShapeConverter.java

```diff
@@ -83,74 +91,74 @@
 
                         for (int i = 0; i < aPolygon.getNumberOfParts(); i++) {
                             Polygon polygon = new Polygon(map);
-                            if (metadata != null) {
-                                metadata.setStringCharset(Charset.defaultCharset());
-
-                                polygon.setSnippet(metadata.toMap().toString());
-                                polygon.setTitle(getSensibleTitle(polygon.getSnippet()));
-                            }
 
                             PointData[] points = aPolygon.getPointsOfPart(i);
                             List<GeoPoint> pts = new ArrayList<>();
 
                             for (PointData p : points) {
-                                GeoPoint pt = new GeoPoint(p.getY(), p.getX());
+                                GeoPoint pt = fixOutOfRange(new GeoPoint(p.getY(), p.getX()));
                                 pts.add(pt);
                             }
                             pts.add(pts.get(0));    //force the polygon to close
 
-                            polygon.setPoints(pts);
-                            final BoundingBox boundingBox = polygon.getBounds();
-                            polygon.setSubDescription(boundingBox.toString());
-                            folder.add(polygon);
+                            polygon.setPoints(pts); //points out of range should be fixed before this line.
+
+                            shapeMetaSetter.set(metadata, polygon);
```

Comment:
Just `shapeMetaSetter.set(metadata, polygon)`: the other three lines `polygon.setSnippet(shapeMetaSetter.getSnippet())`, `polygon.setTitle(shapeMetaSetter.getTitle())` and `polygon.setSubDescription(shapeMetaSetter.getSubDescription())` plus the fillPaint and outlinePaint will be performed inside `shapeMetaSetter.set(metadata, polygon)`.